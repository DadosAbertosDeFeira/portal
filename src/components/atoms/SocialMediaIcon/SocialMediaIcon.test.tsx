import { render } from "@testing-library/react";

import { SocialMediaIcon, type SocialMediaIconProps } from ".";

describe("SocialMediaIcon", () => {
  const makeSut = ({
    type = "facebook",
    ...props
  }: Partial<SocialMediaIconProps>) => {
    return render(<SocialMediaIcon type={type} {...props} />);
  };

  it("should render facebook type correctly", () => {
    const { asFragment } = makeSut({ type: "facebook" });

    expect(asFragment).toMatchSnapshot();
  });

  it("should render github type correctly", () => {
    const { asFragment } = makeSut({ type: "github" });

    expect(asFragment).toMatchSnapshot();
  });

  it("should render instagram type correctly", () => {
    const { asFragment } = makeSut({ type: "instagram" });

    expect(asFragment).toMatchSnapshot();
  });

  it("should render twitter type correctly", () => {
    const { asFragment } = makeSut({ type: "twitter" });

    expect(asFragment).toMatchSnapshot();
  });
});
