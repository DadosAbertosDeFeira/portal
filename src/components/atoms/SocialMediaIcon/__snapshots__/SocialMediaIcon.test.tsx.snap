// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialMediaIcon should render facebook type correctly 1`] = `[Function]`;

exports[`SocialMediaIcon should render github type correctly 1`] = `[Function]`;

exports[`SocialMediaIcon should render instagram type correctly 1`] = `[Function]`;

exports[`SocialMediaIcon should render twitter type correctly 1`] = `[Function]`;
